src/sentry/workflow_engine/endpoints/serializers.py

```diff
@@ -167,3 +180,35 @@ def serialize(self, obj: Detector, attrs: Mapping[str, Any], user, **kwargs) ->
             "conditionGroup": attrs.get("condition_group"),
             "config": attrs.get("config"),
         }
+
+
+@register(Workflow)
+class WorkflowSerializer(Serializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        attrs: MutableMapping[Workflow, dict[str, Any]] = defaultdict(dict)
+        condition_groups = list(
+            DataConditionGroup.objects.filter(
+                id__in=[w.when_condition_group_id for w in item_list if w.when_condition_group_id]
+            )
+        )
+        condition_group_map = {
+            str(group.id): serialized
+            for group, serialized in zip(condition_groups, serialize(condition_groups, user=user))
+        }
+        for item in item_list:
+            attrs[item]["condition_group"] = condition_group_map.get(
+                str(item.when_condition_group_id)
+            )
+        return attrs
+
+    def serialize(self, obj: Workflow, attrs: Mapping[str, Any], user, **kwargs) -> dict[str, Any]:
+        # WHAT TO DO ABOUT CONFIG?
```

Comment:
This is a good question 🤔 Cathy added enforcement of the schemas in https://github.com/getsentry/sentry/pull/81979/files and I was working on defining a schema for the detector config [here](https://github.com/getsentry/sentry/pull/82216) that I need to get back to but we haven't made others yet afaik. Wanna bring it up in office hours tomorrow?